ReactViewResources/Loader.ts

```diff
@@ -117,36 +122,41 @@ export function loadStyleSheet(stylesheet: string): void {
     innerLoad();
 }
 
-export function loadPlugins(plugins: string[][], mappings: Dictionary<string>): void {
+export function loadPlugins(plugins: string[][]): void {
     async function innerLoad() {
         try {
             await BootstrapTask.promise;
 
-            if (mappings) {
-                let paths = Object.assign(getRequirePaths(), mappings);
-                require.config({
-                    paths: paths
-                });
-            }
-
             if (plugins && plugins.length > 0) {
                 // load plugin modules
                 let pluginsPromises: Promise<void>[] = [];
                 plugins.forEach(m => {
-                    const ModuleName = m[0];
-                    const NativeObjectFullName = m[1]; // fullname with frame name included
+                    const ModuleName: string = m[0];
+                    const NativeObjectFullName: string = m[1]; // fullname with frame name included
                     const NativeObjectName = m[2]; // name without frame name
+                    const MainJsSource: string = m[3];
+
+                    let dependencyJsSources: string[] = m.length > 2 ? m.slice(3) : [];
+
                     pluginsPromises.push(new Promise<void>((resolve, reject) => {
-                        CefSharp.BindObjectAsync(NativeObjectFullName, NativeObjectFullName).then(() => {
+                        CefSharp.BindObjectAsync(NativeObjectFullName, NativeObjectFullName).then(async () => {
                             window[NativeObjectName] = window[NativeObjectFullName]; // create alias to the original name, since views are not aware of the fullname
-                            require([ModuleName], (Module: any) => {
-                                if (Module) {
-                                    Modules[ModuleName] = Module.default;
-                                    resolve();
-                                } else {
-                                    reject(`Failed to load '${ModuleName}' (might not be a module)`);
-                                }
-                            });
+
+                            if (ModuleName) {
+
+                                // plugin dependency js sources
+                                let dependencySourcesPromises: Promise<void>[] = [];
```

Comment:
You can move the dependencies promises load outside of the BindObjectAsync, right, and leave just the await Promise.all(dependencySourcesPromises); here?